### PR TITLE
GitRev improvements

### PR DESCRIPTION
**Pull request by csware**
- [x] Improved error handling
- [x] Usage of libgit2 for GitRev
- [x] Code cleanup
- [x] Usage of libgit2 for fetching file lists and diffs, but disabled for default as this will need additional memory for the libgit2 caches in addition to the libgit data

### CODE REVIEW

File: src/Git/GitRevLoglist.cpp [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+		git->CheckAndInitDll();
+		GIT_COMMIT commit = { 0 };
+		GIT_COMMIT_LIST list;
+		GIT_HASH   parent;
+
+		CAutoLocker lock(g_Git.m_critGitDllSec);
+
+		try
+		{
+			if (git_get_commit_from_hash(&commit, m_CommitHash.m_hash))
```
Comment (csware) [inline src/Git/GitRevLoglist.cpp]: As we're requesting a new commit object this is also a candidate for using libgit2

File: src/Git/GitRevLoglist.cpp [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
```
Comment (csware) [inline src/Git/GitRevLoglist.cpp]: Also a potential candidate for using libgit2

**Comment by csware** [comment]
@ch3cooli You added code for diffing using libgit2, are there any known issues why we haven't activated that?

**Comment by ch3cooli** [comment]
Not tested the libgit2 diff function after one year's time.

**Pull request merged**